js/src/core/templatablepattern.js

```diff
@@ -63,8 +63,9 @@ function TemplatablePattern(input_scanner, parent) {
     django: pattern.starting_with(/{%/).until_after(/%}/),
     django_value: pattern.starting_with(/{{/).until_after(/}}/),
     django_comment: pattern.starting_with(/{#/).until_after(/#}/),
-    smarty: pattern.starting_with(/{(?=[^}\s\n])/).until_after(/}/),
-    smarty_comment: pattern.starting_with(/{\*/).until_after(/\*}/)
+    smarty: pattern.starting_with(/{(?=[^}{\s\n])/).until_after(/[^\s\n]}/),
+    smarty_comment: pattern.starting_with(/{\*/).until_after(/\*}/),
+    smarty_literal: pattern.starting_with(/{literal}/).until_after(/{\/literal}/)
```

Comment:
According to docs, literals are used around Javascript or stylesheet. So allowing formatting inside literal would be better.

Reply:
Yes, however this is basic support.  Feel free to file an issue for that and do the work in a future PR.

Reply:
Okay, let's do that later.